upgrade/upgrade-scripts/execute

```diff
@@ -125,36 +132,112 @@ apt_get install -y "delphix-platform-$platform" ||
 	die "failed to install delphix-platform"
 
 #
-# Now that we know we have a delphix-platform-<platform> packages
-# installed, go ahead and install delphix-entire-<platform>.
+# To enable the use of "autoremove" later in this script to remove all
+# packages that are no longer required after the upgrade, we need to
+# mark all packages currently installed as "auto" installed. This way,
+# after explicitly installing all packages for the new "delphix-entire"
+# version, the only "auto" installed packages still remaining on the
+# system will be ones that aren't required by the new "delphix-entire".
+#
+dpkg-query -Wf '${Package}\n' | xargs apt-mark auto ||
+	die "failed to mark packages as 'auto' installed"
+
+#
+# While we marked all packages to "auto" above, so they'd be removed
```

Comment:
Do we know the amount of extra space that those kernel packages would consume? I presume that they would get cleaned-up from time to time by not-in-place upgrades, but it would be a good idea to know if we need to implement some cleanup beyond a reasonable number of kernels.

Reply:
I don't know off-hand, but I can find out for the currently install package(s).

Reply:
I don't really know if this is accurate, or what the units are:
```
$ apt-cache depends delphix-kernel-4.15.0-1032-aws | grep Depends | awk '{print $2}' | xargs dpkg-query -Wf '${Installed-Size}\n' | awk '{s+=$1} END {printf "%.0f\n", s}'
96171
```

So here's another (more complicated) attempt:
```
$ apt-cache depends delphix-kernel-4.15.0-1032-aws | grep Depends | awk '{print $2}' | xargs -n 1 dpkg -L | while read path; do [[ -f $path ]] && echo $path; done | xargs -n 1 du -b | sort -n
31      /usr/share/doc/linux-aws/changelog.gz
178     /usr/share/doc/connstat-module-4.15.0-1032-aws/changelog.Debian.gz
383     /usr/share/doc/connstat-module-4.15.0-1032-aws/copyright
436     /etc/depmod.d/zfs-modules-4.15.0-1032-aws.conf
530     /usr/share/doc/linux-tools-aws/changelog.gz
684     /usr/share/doc/zfs-modules-4.15.0-1032-aws/README.Debian
1292    /usr/share/doc/linux-image-4.15.0-1032-aws/copyright
1598    /usr/share/doc/linux-aws/copyright
1598    /usr/share/doc/linux-tools-aws/copyright
4507    /usr/share/doc/zfs-modules-4.15.0-1032-aws/changelog.Debian.gz
7848    /lib/modules/4.15.0-1032-aws/kernel/net/connstat/connstat.ko
40358   /usr/share/doc/linux-image-4.15.0-1032-aws/changelog.Debian.gz
53390   /usr/share/doc/zfs-modules-4.15.0-1032-aws/copyright
240112  /lib/modules/4.15.0-1032-aws/extra/avl/zavl.ko
1055592 /lib/modules/4.15.0-1032-aws/extra/unicode/zunicode.ko
1183384 /lib/modules/4.15.0-1032-aws/extra/nvpair/znvpair.ko
3629312 /lib/modules/4.15.0-1032-aws/extra/zcommon/zcommon.ko
4466232 /lib/modules/4.15.0-1032-aws/extra/spl/spl.ko
7919472 /boot/vmlinuz-4.15.0-1032-aws
8076512 /lib/modules/4.15.0-1032-aws/extra/lua/zlua.ko
12367512        /lib/modules/4.15.0-1032-aws/extra/icp/icp.ko
59238592        /lib/modules/4.15.0-1032-aws/extra/zfs/zfs.ko
```
and summing these up, gives:
```
$ apt-cache depends delphix-kernel-4.15.0-1032-aws | grep Depends | awk '{print $2}' | xargs -n 1 dpkg -L | while read path; do [[ -f $path ]] && echo $path; done | xargs -n 1 du -b | sort -n | awk '{s+=$1} END {printf "%.0f\n", s}'
98289553
```
So that's just about 93MB, and doesn't include the kernel's debug symbols package.

The information for the kernel's "dbgsym" package is here:
```
 dpkg -L linux-image-4.15.0-1032-aws-dbgsym | while read path; do [[ -f $path ]] && echo $path; done | xargs -n 1 du -b | sort -n | awk '{s+=$1} END {printf "%.0f\n", s}'
3636521147
```
So, about 3.3GB just for the kernels debug symbols package.

I think it'd be reasonable to take a conservative approach and say we'll need at least 4GB to hold the kernel packages for each old release that happens to get installed on the system (assuming we keep the old kernel "dbgsym" package).

Reply:
IMO, I'd like to leave this as-is, and then if we determine keeping all kernel versions installed is a problem (e.g. due to the space consumption), we can address it later.

Reply:
I'm fine with addressing that later. Let's open a bug for that so that we can track this. 

Regarding debug symbols, I don't believe we currently install them on the external-standard image, do we? I think they would make the upgrade image double in size.

Reply:
We install it on the external images. We add the "dbgsym" packages in the "minimal-common" role, so it should be in all variants. We want them all variants so we can use kernel debug tools like `crash` (at least, that's been my understanding).